units/provisioning_agent_deploy.go

```diff
@@ -374,3 +385,22 @@ func (j *agentDeployJob) startAgentOnRemote(ctx context.Context, settings *everg
 
 	return nil
 }
+
+func CheckExternallyTerminated(ctx context.Context, env evergreen.Environment, h *host.Host) (bool, error) {
```

Comment:
Can this share code with the host-monitoring-external-state-check job?

Reply:
done.